discussion_agents/cog/functional/react.py

```diff
@@ -81,13 +88,39 @@ def _is_halted(
         scratchpad (str): The scratchpad content.
         max_tokens (int): Maximum allowed token count.
         enc (Encoding): The encoder to calculate token length.
-
+        examples (str): The example of the output prompt.
+        instruction (str): The template of the prompt.
     Returns:
         bool: True if the operation should be halted, False otherwise.
     """
     over_max_steps = step_n > max_steps
     over_token_limit = (
-        len(enc.encode(_build_agent_prompt(question=question, scratchpad=scratchpad)))
+        len(enc.encode(_build_agent_prompt(question=question, scratchpad=scratchpad, examples=examples, instruction=instruction)))
         > max_tokens
     )
     return finished or over_max_steps or over_token_limit
+
+
+
+def check_type(examples: str = None) -> str:
+    """String check of examples to classify benchmark."""
+    lines = examples.split('\n')
+    lines = [line for line in lines if line.strip()] 
+    checkword = lines[0].split()[0]
+    if checkword == 'Question:':
+        return HOTPOTQA
+    else:
+        checkword = lines[0].split()[0]
+        if checkword == 'Claim:':
+            return FEVER
+        line = lines[2]
+        if 'Your task is to:' in line :
+            return ALFWORLD
+    return ValueError('Wrong Examples')
+
+
+def process_ob(ob):
```

Comment:
check_type and process_ob should both be prefixed by a `_` if they are meant to be used internally